code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,918 @@
+
+/*
+Warping extracts crossbreed
+put up a rune with bluespace effects, lots of those runes are fluff or act as a passive buff, others are just griefind tools.
+
+*/
+
+/obj/item/slimecross/warping
+	name = "warped extract"
+	desc = "It just won't stay in place."
+	icon_state = "warping"
+	effect = "warping"
+	colour = "grey" ///default color of the crossbreed
+	var/obj/effect/warped_rune/runepath ///what runes will be drawn depending on the crossbreed color
+	var/warp_charge = 1 /// the number of "charge" a bluespace crossbreed start with
+	var/max_charge = 1 ///max number of charge, might be different depending on the crossbreed (all crossbreed have 1 max charge for now)
+	var/storing_time = 15 ///time it takes to store the rune back into the crossbreed
+	var/drawing_time = 15 ///time it takes to draw the rune
+
+/obj/effect/warped_rune
+	name = "warped rune"
+	desc = "An unstable rune born of the depths of bluespace"
+	icon = 'icons/obj/slimecrossing.dmi'
+	icon_state = "greyspace_rune"
+	var/runepath = /obj/effect/warped_rune  ///serves as a way for some specific runes to identify themselves in the turf.TODO: you can just use src dumbass
+	var/storing_time = 5 ///is only used for bluespace crystal erasing as of now
+	anchored = TRUE
+	layer = MID_TURF_LAYER
+	resistance_flags = FIRE_PROOF  //It's only fireproof because of the fire rune and because how do you burn a rune anyway.
```

Comment:
Put the object-specific variables (the ones that start with `var/` at the bottom and the redefined ones (the ones without) at the top, for clarity.